index.js

```diff
@@ -30,6 +32,67 @@ class StoreFacility extends Base {
     return new Autobase(this.store.session(), boostrapKey, baseOpts)
   }
 
+  async clearBeeCache (bee, ckey, maxSize = 1000) {
+    let seqCheckpoint = await bee.core.getUserData(ckey)
+    seqCheckpoint = seqCheckpoint ? convFromBin(seqCheckpoint, 'number') : 0
+    const map = new LimitedMap(maxSize)
+
+    for await (const entry of bee.createHistoryStream({ gt: seqCheckpoint, lt: bee.core.length - 1 })) {
+      if (entry.type === 'del') {
+        await bee.core.setUserData(ckey, convIntToBin(entry.seq))
+        await bee.core.clear(entry.seq)
+        continue
+      }
+
+      let seq = map.get(entry.key)
+      if (!seq) {
+        const latest = await bee.get(entry.key)
+        seq = latest?.seq ?? -1
+        map.set(entry.key, seq)
+      }
+
+      if (seq === -1 || entry.seq !== seq) {
+        await bee.core.setUserData(ckey, convIntToBin(entry.seq))
+        await bee.core.clear(entry.seq)
+      }
+    }
+  }
+
+  async putAndClear (bee, key, value, opts = {}) {
+    const shouldCheckIfUpdateHappened = !!opts.cas
+    const existingEntry = await bee.get(key)
+    await bee.put(key, value, opts)
+
+    if (!existingEntry?.seq) {
+      return
+    }
+    if (shouldCheckIfUpdateHappened) {
+      const updatedEntry = await bee.get(key)
+      if (updatedEntry?.seq === existingEntry?.seq) {
+        // cas returned false and so don't need to clear existing entry
+        return
+      }
+    }
+    await bee.core.clear(existingEntry.seq)
+  }
```

Comment:
```suggestion
  async putAndClear (bee, key, value, putOpts = undefined, getOpts = undefined) {
    const shouldCheckIfUpdateHappened = !!putOpts?.cas
    const existingEntry = await bee.get(key, getOpts)
    await bee.put(key, value, putOpts)

    if (!existingEntry?.seq) {
      return
    }
    if (shouldCheckIfUpdateHappened) {
      const updatedEntry = await bee.get(key, getOpts)
      if (updatedEntry?.seq === existingEntry?.seq) {
        // cas returned false and so don't need to clear existing entry
        return
      }
    }
    await bee.core.clear(existingEntry.seq)
  }
```

Reply:
any particular reason you wanna use undefined instead of {}?

Reply:
okay, since undefined is the default value for non-existing params anyway, skipped defining ..Opts= undefined in function definition